File: Baekjoon/step by step/06_intensification/1316.cpp
// 1316: 그룹 단어 체커

#include <iostream>
#include <string>
#include <unordered_map>

int main() {
  int group_cnt = 0, N = 0;
  std::cin >> N;

  for (int i=0; i<N; i++) {
    std::string word;
    std::cin >> word;

    std::unordered_map<char, bool> check_dict;
    bool is_group = true;

    
    for (int j=0; j<word.length(); j++) {
      if (check_dict.find(word.at(j)) == check_dict.end()) {
        check_dict[word.at(j)] = true;
        continue;
      }

      if (check_dict[word.at(j)] && word.at(j) != word.at(j-1)) {
        is_group = false;
        break;
      }      
    }

    group_cnt += is_group ? 1:0;
  }

  std::cout << group_cnt;

  return 0;
}

File: Baekjoon/step by step/06_intensification/1316.py
# 1316: 그룹 단어 체커

# 1. N 입력
N = int(input())
group_cnt = 0

for i in range(N):
  # 2. 단어 입력
  word = input()

  # 3. 단어 체크
  check_dict = {}
  is_group = True
  for i in range(len(word)):
    # 3.2 비연속 중복 입력인 경우
    if (check_dict.get(word[i], False) and word[i] != word[i-1]):
      is_group = False
      break

    check_dict[word[i]] = True
  
  group_cnt += 1 if is_group else 0

print(group_cnt)
